benchmarks/makeARGB.py

```diff
@@ -1,72 +1,130 @@
+        uint16_data = {
+            "data": xp.random.randint(100, 4500, size=size).astype("uint16"),
+            "levels": [250, 3000],
         }
-
-        self.uint8_data = {
-            'data': np.random.randint(0, 255, size=size).astype('ubyte'),
-            'levels': [20, 220],
+        uint8_data = {
+            "data": xp.random.randint(0, 255, size=size).astype("ubyte"),
+            "levels": [20, 220],
         }
-
-        self.c_map = np.array([
-            [-500., 255.],
-            [-255., 255.],
-            [0., 500.],
-        ])
-
-        self.uint8_lut = np.zeros((256, 4), dtype='ubyte')
+        c_map = xp.array([[-500.0, 255.0], [-255.0, 255.0], [0.0, 500.0]])
+        uint8_lut = xp.zeros((256, 4), dtype="ubyte")
         for i in range(3):
-            self.uint8_lut[:, i] = np.clip(np.linspace(self.c_map[i][0], self.c_map[i][1], 256), 0, 255)
-        self.uint8_lut[:, 3] = 255
-
-        self.uint16_lut = np.zeros((2 ** 16, 4), dtype='ubyte')
+            uint8_lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 256), 0, 255)
+        uint8_lut[:, 3] = 255
+        uint16_lut = xp.zeros((2 ** 16, 4), dtype="ubyte")
         for i in range(3):
-            self.uint16_lut[:, i] = np.clip(np.linspace(self.c_map[i][0], self.c_map[i][1], 2 ** 16), 0, 255)
-        self.uint16_lut[:, 3] = 255
+            uint16_lut[:, i] = xp.clip(xp.linspace(c_map[i][0], c_map[i][1], 2 ** 16), 0, 255)
+        uint16_lut[:, 3] = 255
+        return float_data, uint16_data, uint8_data, uint16_lut, uint8_lut
 
 
-def make_test(dtype, use_levels, lut_name, func_name):
+def make_test(dtype, use_cupy, use_levels, lut_name, func_name):
     def time_test(self):
-        data = getattr(self, dtype + '_data')
-        makeARGB(
-            data['data'],
-            lut=getattr(self, lut_name + '_lut', None),
-            levels=use_levels and data['levels'],
-        )
+        data = getattr(self, dtype + "_data")
+        levels = data["levels"] if use_levels else None
+        lut = getattr(self, lut_name + "_lut", None) if lut_name is not None else None
+        for _ in range(10):
+            img_data = data["data"]
+            output = self.output
+            if use_cupy:
+                if not cp:
+                    raise ValueError("Cannot test cupy without it being installed")
+                img_data = cp.asarray(img_data)
+                output = self.cupy_output
+            makeARGB(
+                img_data, lut=lut, levels=levels, output=output,
+            )
+            if use_cupy:
+                output.get(out=self.output)
 
     time_test.__name__ = func_name
     return time_test
 
 
-for dt in ['float', 'uint16', 'uint8']:
-    for levels in [True, False]:
-        for ln in [None, 'uint8', 'uint16']:
-            name = f'time_makeARGB_{dt}_{"" if levels else "no"}levels_{ln or "no"}lut'
-            setattr(TimeSuite, name, make_test(dt, levels, ln, name))
+for cupy in [True, False]:
```

Comment:
Can we make `cupy` here only have a `True` option  if cupy is installed?

Reply:
I hadn't imagined we'd have separate GPU and CPU benchmark systems, and I definitely want to see both measurements.

Reply:
oh I agree, we should have one benchmark system, for both.  What happens here with `True` for `cupy` is if you don't have `cupy` installed/available, half the tests just fail.    The idea here is that those tests don't run at all if cupy is not installed.

Reply:
Oh! Yes, I read your comment backwards ( or maybe sideways ). Thanks for elucidating; I agree now.